include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -1801,34 +1805,52 @@ namespace internal
           for (unsigned int my_q = 0; my_q < mapping_info.face_data.size();
                ++my_q)
             {
+#ifndef DEAL_II_WITH_SIMPLEX_SUPPORT
               // currently only non-hp case...
-              Assert(mapping_info.face_data[my_q].descriptor.size() == 1,
-                     ExcNotImplemented());
+              AssertDimension(mapping_in.size(), 1);
+              AssertDimension(mapping_info.face_data[my_q].descriptor.size(),
+                              1);
+#endif
+
+              const unsigned int fe_index =
+                active_fe_index.size() > 0 ?
+                  active_fe_index[faces[face].cells_interior[0]] :
+                  0;
```

Comment:
This PR does not enable hp + DG in `MatrixFree` but is targeting the case that `Mappings` and `Quadrature` rules match at faces like in the case of mixed meshes. 

This is why I am selecting here the `active_fe_index` of the inner cell.